1_The_Physical/Level_2_Stateful_Guess_Point/Shared/Student/src/domain/core/Validator.spec.ts

```diff
@@ -0,0 +1,149 @@
+import { StringValidator } from "./Validator";
+
+describe("Validator", () => {
+  class TestValidator extends StringValidator {
+    protected requiredMessage = "Test is required";
+    protected pattern = /^[a-zA-Z]+$/;
+    protected patternMessage = "Test must contain only letters";
+    protected minLength = 4;
+    protected minLengthMessage = "Test must be at least 4 characters long";
+    protected maxLength = 10;
+    protected maxLengthMessage = "Test must be at most 10 characters long";
+  }
+
+  describe("validateRequired", () => {
+    it("should return an error if props is empty", () => {
+      const validator = new TestValidator();
+
+      const result = validator.validate(null as any);
```

Comment:
I believe you can rely on TS types (make sure you use `strict: true` in `tsconfig.json` though. So, you don't need to hack it, such errors would be caught during compilation time. Moreover, you could just pass and empty string.